src/constraint_aware_append.c

```diff
@@ -563,10 +563,17 @@ static CustomPathMethods constraint_aware_append_path_methods = {
 };
 
 Path *
-ts_constraint_aware_append_path_create(PlannerInfo *root, Hypertable *ht, Path *subpath)
+ts_constraint_aware_append_path_create(PlannerInfo *root, Hypertable *ht, Path *subpath,
+									   RelOptInfo *rel)
 {
 	ConstraintAwareAppendPath *path;
 
+#if !(PG96 || PG10)
```

Comment:
I guess this could be a short-term fix, but it seems we are forgoing potential optimizations for partitioning in order to do runtime pruning. 

Also wondering if this is the right place to turn this off as it doesn't seem to be this node's business to make this decision. Seems like it should be a more top-level decision by the planner, e.g., at the same level it is deciding whether it should optimize append. Now this seems like a hidden non-obvious side-effect.

One other thing to note here is that PG partitioning has its own runtime pruning, so wondering if there's a way to hook into that in the future rather than doing our own thing.